### DISCUSSION TITLE
Analysing the Overcharge / discharge behaviour in PyBaMM Model

### DISCUSSION THREAD

**Discussion by Akila1993** [pybamm-team/PyBaMM#2799] [Q&A]
1. Is there is a possibility run the PyBaMM model for overcharge or over discharge scenarios. Is it possible with only experiment class.

2. Also can we inject different c-rates for different frequency values, therefore we can obtain the electrochemical impedance spectroscopy for certain model that we are interested in.

**Comment by brosaplanella** [comment]
1. What do you mean by overcharge and overdischarge? Going beyond the recommended voltage limits?
2. You can do that by specifying the current function (see [this notebook](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/parameterization/change-input-current.ipynb) for an example).

**Reply by Akila1993** [reply]
Thank you for your response.

1. Yes it is referring to the overcharge/discharge fault. According to following definitions. 

Overcharge: The behavior of continuing to charge the battery after the battery is fully charged.
Over-discharge: The behavior of continuing to discharge the battery after the battery reached the discharge cut-off voltage.

Is there is a possibility run the PyBaMM model for these scenarios

**Reply by brosaplanella** [reply]
I think that specifying and experiment like `Discharge at 1C until 2 V` should work. If it doesn't, please share a minimum working example and the full error log here.

**Reply by Akila1993** [reply]
I just tried the example in tutorial to run to 2V and following warning appeared.

experiment = pybamm.Experiment(
    [
        ("Discharge at 1C until 2 V",
        "Rest for 1 hour",
        "Charge at 1 A until 4.1 V",
        "Hold at 4.1 V until 50 mA",
        "Rest for 1 hour"),
    ] * 3
)

model = pybamm.lithium_ion.DFN()

sim = pybamm.Simulation(model, experiment=experiment)

sim.solve()
sim.plot()

****Error Message:**** 

023-03-23 18:06:43.466 - [WARNING] callbacks.on_experiment_infeasible(233): 

	Experiment is infeasible: 'event: Minimum voltage' was triggered during 'Discharge at 1C until 2 V'. The returned solution only contains up to step 1 of cycle 1. 
WARNING:pybamm.logger:

	Experiment is infeasible: 'event: Minimum voltage' was triggered during 'Discharge at 1C until 2 V'. The returned solution only contains up to step 1 of cycle 1. 
t
0.00

**Reply by Akila1993** [reply]
@brosaplanella Is there is any comment for above results?

**Reply by brosaplanella** [reply]
Which parameters are you using? I suspect the parameter set you use cannot be discharged until 2 V as it reaches negative concentrations.

**Reply by Akila1993** [reply]
@brosaplanella Can we update the lower or upper cut-off voltage level of a any parameter set and run this kind of experiment ?

**Reply by brosaplanella** [reply]
The cut off voltages do not get used in the experiments (they get overriden with what the experiment says). In this case it seems an issue with negative concentrations.

**Reply by Akila1993** [reply]
@brosaplanella If so what is the possibility to run and overcharge or discharge scenario, can we update the concentration values, or is there is any other way.

**Reply by brosaplanella** [reply]
Yes, you can change the parameters, but note that it is not straightforward to find which parameters will allow you to go all the way down to 2 V. Apparently it works with Chen2020 but haven't tested it much.
